### ISSUE TITLE
-Wunneeded-internal-declaration in drivers/mfd/cros_ec_dev.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#179]
```make
drivers/mfd/cros_ec_dev.c:509:40: warning: variable 'cros_ec_id' is not needed and will not be emitted [-Wunneeded-internal-declaration]
static const struct platform_device_id cros_ec_id[] = {
                                       ^
1 warning generated.
```

I think that `cros_ec_id` should be added to `cros_ec_dev_driver` with the `id_table` member. Patch sent under that assumption: https://lore.kernel.org/lkml/20180927033316.9784-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/pub/scm/linux/kernel/git/lee/mfd.git/commit/?id=de6f25ddc4289596c9fbaec91be82d14fff4eeee

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=6eb357844373f833f085e63e6c8c026a61568c57

**Comment by nickdesaulniers** [comment]
possibly also fixed on the llvm side by #232
